app/models/concerns/progress.rb

```diff
@@ -0,0 +1,17 @@
+module Progress
+  extend ActiveSupport::Concern
+
+  included do
+    before_save :parent, unless: :parent_id?
+    belongs_to :parent, class_name: 'Indicator', optional: true
+  end
+
+  def parent
+    super || assign_attributes(parent: parent_content&.progress_for(user, organization))
```

Comment:
Why are we using assign_attributes instead of update! ?

Reply:
I believe the before_save callback would cause it to go on an endless loop since update! persists the object :thinking:

Reply:
True!